civis/ml/_model.py

```diff
@@ -134,7 +134,8 @@ def _retrieve_file(fname, job_id, run_id, local_dir, client=None):
     """Download a Civis file using a reference on a previous run"""
     file_id = cio.file_id_from_run_output(fname, job_id, run_id, client=client)
     fpath = os.path.join(local_dir, fname)
-    os.makedirs(os.path.dirname(fpath), exist_ok=True)
+    if not os.path.exists(os.path.dirname(local_dir)):
```

Comment:
The `os.path.dirname(local_dir)` here will check a different path than the `os.makedirs` below makes.
```
In [32]: tdir = tempfile.TemporaryDirectory()

In [33]: print(tdir.name, '\n', os.path.dirname(tdir.name))
/var/folders/jn/z8wk4r050bn_zsxmyrhb4gsw0000gn/T/tmpirbgopmi
 /var/folders/jn/z8wk4r050bn_zsxmyrhb4gsw0000gn/T
```

How about
```python
output_dir = os.path.dirname(fpath)
if not os.path.exists(output_dir):
    os.makedirs(output_dir)
```